lib/RGBot.js

```diff
@@ -583,196 +1098,360 @@ const RGBot = class {
+            } else if (a.name === b.name) {
+                return 0
+            } else {
+                return 1
+            }
+        }).shift()
+    }
+
     /**
      * Dig the given Block.
-     * This will equip the most appropriate tool in the Bot's inventory for this Block type.
+     * This will equip the most appropriate tool in the bot inventory for this Block type.
      * @param {Block} block The Block instance to dig
      * @returns {Promise<boolean>} Whether the Block was successfully dug
      */
     async digBlock(block) {
-        let result = false;
         if (!block) {
-            console.error(`digBlock: Block was null or undefined`);
-        }
-        else if (this.#bot.digTime(block) >= 100_000) {
-            console.error(`digBlock: No capable tool to dig block ${this.getEntityName(block)}`);
-        }
-        else {
-            await this.equipBestHarvestTool(block);
-            const checkForInfiniteDig = async (reason) => {
-                if (reason === 'block_updated' || reason === 'dig_error') {
-                    // if Bot is still digging but the target block no longer exists then stop the Bot
-                    if (this.#bot.pathfinder.isMining() && !this.#bot.targetDigBlock) {
-                        this.#log('Cancelling current dig because target block no longer exists.');
-                        this.#bot.stopDigging();
-                        try {
-                            this.#bot.pathfinder.setGoal(null);
-                            this.#bot.pathfinder.stop();
-                        } catch(err) {
-                            this.#log(`Caught expected goal changed error while stopping dig path`);
+            this.#logWarn(`digBlock: Block was null or undefined`);
+        } else {
+            const bestTool = this.bestHarvestTool(block);
+            if (bestTool.digTime < Infinity) {
+                try {
+                    await this.#bot.equip(bestTool.tool, 'hand');
+                    const checkForInfiniteDig = async (reason) => {
+                        if (reason === 'block_updated' || reason === 'dig_error') {
+                            // if Bot is still digging but the target block no longer exists then stop the Bot
+                            if (this.#bot.pathfinder.isMining() && !this.#bot.targetDigBlock) {
+                                this.#logDebug('Cancelling current dig because target block no longer exists.');
+                                this.#bot.stopDigging();
+                                try {
+                                    this.#bot.pathfinder.setGoal(null);
+                                    this.#bot.pathfinder.stop();
+                                } catch (err) {
+                                    this.#logDebug(`Caught expected goal changed error while stopping dig path`);
+                                }
+                            }
                         }
                     }
```

Comment:
I feel like this could be a helper function?

Reply:
Its usage is very specific to how we many the event listeners inside digBlock.  Following in the vein of 'keep it simple', this seems a very complex concept to open up to users.